src/modules/date/index.ts

```diff
@@ -1108,7 +1114,14 @@ export class DateModule extends SimpleDateModule {
       type = 'wide';
     }
 
-    return this.faker.helpers.arrayElement(source[type]);
+    const values = source[type];
+
+    if (values == null) {
+      // This should never happen due to the checks above, but TS doesn't know that
+      throw new FakerError(`No month values found for type '${type}'.`);
+    }
+
+    return this.faker.helpers.arrayElement(values);
```

Comment:
Can't we simply this by letting `arrayElement` handle the "error tha is never possible"?

```suggestion
    return this.faker.helpers.arrayElement(source[type] ?? []);
```

Reply:
Sure, but then there wouldnt be a hint which data are actually missing.

Reply:
Will be considered by solving https://github.com/faker-js/faker/pull/2435#discussion_r1349312186.

Reply:
Fixed